contracts/NodeDelegator.sol

```diff
@@ -145,82 +158,104 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
+    /// @notice Fetches the balance of all supported assets in this NodeDelegator and the underlying EigenLayer.
+    /// This includes LSTs and WETH.
+    /// @return assets a list of assets addresses
+    /// @return assetBalances the balances of the assets in the node delegator and EigenLayer
     function getAssetBalances()
         external
         view
         override
         returns (address[] memory assets, uint256[] memory assetBalances)
     {
-        address eigenlayerStrategyManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_STRATEGY_MANAGER);
-
-        (IStrategy[] memory strategies,) =
-            IEigenStrategyManager(eigenlayerStrategyManagerAddress).getDeposits(address(this));
+        assets = lrtConfig.getSupportedAssetList();
+        assetBalances = new uint256[](assets.length);
 
-        uint256 strategiesLength = strategies.length;
-        assets = new address[](strategiesLength);
-        assetBalances = new uint256[](strategiesLength);
+        for (uint256 i = 0; i < assets.length;) {
+            (uint256 ndcAssets, uint256 eigenAssets) = getAssetBalance(assets[i]);
+            assetBalances[i] = ndcAssets + eigenAssets;
 
-        for (uint256 i = 0; i < strategiesLength;) {
-            assets[i] = address(IStrategy(strategies[i]).underlyingToken());
-            assetBalances[i] = IStrategy(strategies[i]).userUnderlyingView(address(this));
             unchecked {
                 ++i;
             }
         }
     }
 
-    /// @dev Returns the balance of an asset that the node delegator has deposited into the strategy
-    /// @param asset the asset to get the balance of
-    /// @return stakedBalance the balance of the asset
-    function getAssetBalance(address asset) external view override returns (uint256) {
-        address strategy = lrtConfig.assetStrategy(asset);
-        if (strategy == address(0)) {
-            return 0;
+    /// @dev Returns the balance of an asset that the node delegator has deposited into an EigenLayer strategy
+    /// or native ETH staked into an EigenPod.
+    /// @param asset the token address of the asset.
+    /// WETH will include any native ETH in this contract or staked in EigenLayer.
+    /// @return ndcAssets assets lying in this NDC contract.
+    /// This includes any native ETH when the asset is WETH.
+    /// @return eigenAssets asset amount deposited in underlying EigenLayer strategy
+    /// or native ETH staked into an EigenPod.
+    function getAssetBalance(address asset) public view override returns (uint256 ndcAssets, uint256 eigenAssets) {
+        ndcAssets += IERC20(asset).balanceOf(address(this));
+
+        if (asset == WETH_TOKEN_ADDRESS) {
+            // Add any ETH in the NDC that was earned from EigenLayer
+            ndcAssets += address(this).balance;
+
+            eigenAssets = stakedButNotVerifiedEth;
+            if (address(eigenPod) != address(0)) {
+                eigenAssets += address(eigenPod).balance;
+            }
```

Comment:
🔴🔴 This calculation of native eth held in Eigenlayer is wrong, and would result in our funds disappearing from our own internal accounting, thus resulting in incorrect, oversized mint amounts.

From the node's point of view, ETH can be at rest in:

1. WETH on the node contract
2. ETH on the node contract
3. ETH on the eigenpod
4. ETH in unverified validators
5. ETH in verified validators.

We don't account for the last amount anywhere, and it's probable that that would be where almost all the ETH will be. This would result in assetBalances too low.

Secondly, the math in our contract (inherited from Kelp) on `stakedButNotVerifiedEth` is incorrect. We should add 32 to it when we stake eth, but we should not subtract validators effective balance from it when it gets verified.

Consider this case. Validator gets slashed a few times and loses 1.5 ETH between staking and verification. stakedButNotVerifiedEth was plus 32 ETH at stake, but gets minus 30.5 ETH at verification. This will result in a wrong value in stakedButNotVerifiedEth, inflating our asset balance by 1.5 ETH forever.

We could have similar bug in our code with rewards earned by the validator before verification, however eigenlayer caps the return value at 32. Still don't like it though.

There's two options for fixing this:

1. Get our verified ETH balance from eigenlayer, and change `stakedButNotVerifiedEth` to only +- exactly 32 increments.
2. Do what Kelp ended up doing, and only plus +32 to stakedButNotVerifiedEth, never subtract, and never use any other method to calculate balances until we add withdraws.

Reply:
Note that it appears that we will need to offline track each withdraw and rewards distribution in order to receive them from eigenlayer? This seems gas expensive in that we have to manually claim each automatic reward distribution from each validator?

Reply:
I think we need to go for the second option for now as I don't think there is way for the NodeDelegator to read the validator balances. I think the only way to get the validator balance to the execution chain is via a validator proof like `verifyWithdrawalCredentials` uses.
I see Kelp have removed `verifyWithdrawalCredentials`. That makes sense as anyone can call `verifyWithdrawalCredentialsAndBalance` on the EigenPod with the validator proofs.
I'll remove `verifyWithdrawalCredentials` from the NodeDelegator on a new PR